src/JWTAuth.php

```diff
@@ -59,11 +59,13 @@ public function authenticate()
     {
         $id = $this->getPayload()->get('sub');
 
-        if (! $this->auth->byId($id)) {
-            return false;
+        $retrieveUser = $this->manager->getRetrieveUserOnAuthentication();
+
+        if ($retrieveUser && $this->auth->byId($id)) {
+            return $this->user();
         }
 
-        return $this->user();
+        return false;
```

Comment:
So when `'retrieve_user_on_authentication' => false`, this method will always do nothing and return false? My concern is that silently breaking a method is more confusing than instructing users not to use it. I'd prefer you to go all out and throw an exception instead (since the method won't ever be useful, you should alert those using it).

If that wasn't your intention or I'm misreading this, let me know. :wink: